### PR TITLE
Deprecate old logger types

### PR DESCRIPTION
**Pull request by markfields**
## Description

Fixes [AB#5446](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5446) and [AB#5447](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5447)

We added "Ext" logging types to be used when instrumenting, to add support for logging flat arrays and objects directly without having to JSON.stringify (the logger does it for you).  This functionality was in new copies of old types, with "Ext" in the name.

This PR deprecates the old types, and includes renaming some other types to align on the "Base" nomenclature for types used in public API surface (which don't include the extended types mentioned above).

## Breaking Changes

This is setting the stage for some breaking changes (removing and renaming several types).

## Reviewer Guidance

I'll remove internal usages of the deprecated APIs in another PR unless CI demands I do it now.

### Question about deprecation -> removal path

I'm planning to remove in `next` immediately, since the burden on consumers is very trivial (renaming the symbol or the package) - and that burden is equivalent to adding exemption comments above each deprecated import.  Any concerns?  I'll check with some internal partners too to clarify.

### CODE REVIEW

File: api-report/telemetry-utils.api.md [hunk 1]
```diff
@@ -211,6 +211,8 @@ export interface ITelemetryEventExt extends ITelemetryPropertiesExt {
 
 // @public
 export interface ITelemetryGenericEventExt extends ITelemetryPropertiesExt {
+    // Warning: (ae-forgotten-export) The symbol "TelemetryEventCategory" needs to be exported by the entry point index.d.ts
```
Comment (markfields) [inline api-report/telemetry-utils.api.md:214]: Need to fix this

File: packages/common/core-interfaces/src/logger.ts [hunk 2]
```diff
@@ -30,8 +51,15 @@ export interface ITaggedTelemetryPropertyType {
 /**
  * JSON-serializable properties, which will be logged with telemetry.
  */
+export type ITelemetryBaseProperties = ITelemetryProperties;
+
+/**
+ * @see {@link ITelemetryBaseProperties}
```
Comment (Josmithr) [inline packages/common/core-interfaces/src/logger.ts:57]: Nit: I think we can still use `inheritDoc` here and add the `deprecated` tag.
```suggestion
 * {@inheritDoc ITelemetryBaseProperties}
```

File: packages/common/core-interfaces/src/logger.ts [hunk 3]
```diff
@@ -30,8 +51,15 @@ export interface ITaggedTelemetryPropertyType {
...
+ *
+ * @deprecated Renamed to ITelemetryBaseProperties
```
Comment (Josmithr) [inline packages/common/core-interfaces/src/logger.ts:59]: Nit (so the generated API docs contain a nice link to the replacement): 
```suggestion
 * @deprecated Renamed to {@link ITelemetryBaseProperties}
```

File: packages/common/core-interfaces/src/logger.ts [hunk 4]
```diff
@@ -15,13 +17,32 @@ export type TelemetryEventCategory = "generic" | "error" | "performance";
  * easily add fields to objects that shouldn't be logged and not realize it's going to be logged.
  * General best practice is to explicitly log the fields you care about from objects.
  */
+export type TelemetryBaseEventPropertyType = TelemetryEventPropertyType;
+
+/**
+ * @see {@link TelemetryBaseEventPropertyType}
```
Comment (Josmithr) [inline packages/common/core-interfaces/src/logger.ts:23]: Nit:
```suggestion
 * {@inheritDocTelemetryBaseEventPropertyType}
```

File: packages/utils/telemetry-utils/src/telemetryTypes.ts [hunk 5]
```diff
@@ -3,11 +3,12 @@
...
 
-import {
-	ITelemetryBaseLogger,
-	LogLevel,
-	TelemetryEventCategory,
-} from "@fluidframework/core-interfaces";
+import { ITelemetryBaseLogger, LogLevel, Tagged } from "@fluidframework/core-interfaces";
+
+/**
+ * Examples of known categories, however category can be any string for extensibility.
```
[the remark this answers is not in the document]
Comment (markfields) [reply packages/utils/telemetry-utils/src/telemetryTypes.ts:9]: Yeah I'm not really sure this comment is useful as-is, I'll change it, esp now that this is moving to the internal instrumentation package (as opposed to core-interfaces).

**Review by Josmithr — APPROVED** [review]

File: packages/common/core-interfaces/src/logger.ts (hunk 2 above)
Comment (markfields) [reply packages/common/core-interfaces/src/logger.ts:57]: Cool I'll try it.  Thanks for all the tips!

**Review by alexvy86 — COMMENTED** [review]
I'd expand the relevant deprecation messages to something like `@deprecated For internal use within FluidFramework, use ITelemetryPerformanceEventExt. No replacement intended for FluidFramework consumers.`. The idea is that at some point (maybe with the work @Josmithr is doing for internal exports) all the `*Ext` types/interfaces become an internal thing that we don't expose, right?

> I'm planning to remove in next immediately, since the burden on consumers is very trivial (renaming the symbol or the package) - and that burden is equivalent to adding exemption comments above each deprecated import. Any concerns? I'll check with some internal partners too to clarify.

In particular for `TelemetryEventCategory` it could mean adding a new dep on telemetry-utils, additional lockfile changes, etc., so I wouldn't remove that one right away. IMO most probably the telemetry-utils dep already exists where it would be necessary, but better play it safe.

File: packages/common/core-interfaces/src/logger.ts (hunk 2 above)
Comment (markfields) [reply packages/common/core-interfaces/src/logger.ts:57]: Ok yeah that got rid of the "undocumented" flag in the api.md.  I wish VSCode respected `@inheritDoc`, is there a plugin that adds that or something?

Comment (Josmithr) [reply packages/common/core-interfaces/src/logger.ts]: It's really inconsistent. It often works, but doesn't always. I _think_ VSCode assumes that there should be no other contents in the comment other than the `@inheritDoc` comment, and when there is it doesn't do the substitution.

Comment (Josmithr) [reply packages/common/core-interfaces/src/logger.ts]: I haven't looked into plugins, but that's an interesting thought.

**Pull request merged**